pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -60,6 +63,7 @@
   private SegmentBuildTimeLeaseExtender _leaseExtender;
   private RealtimeSegmentStatsHistory _statsHistory;
   private final Semaphore _segmentBuildSemaphore;
+  private Map<Integer, Semaphore> _partitionIdToSemaphoreMap = new ConcurrentHashMap<>();
```

Comment:
Please write a block of comment before the declaration of this variable, describing what these semaphores are being used for.
Make it private final.
Also document, that the semaphores will stay in the hash map even if the consuming partitions move to a different host. We expect that these will be a small number of semaphores, but that may be ok. 
It may be useful to add a metric of how many semaphores are there in the map..

Reply:
Comments added.
In terms of the metric, I don't think that's necessary. We brought in a variable called `_acquireConsumerSemaphore` within the LLCSegmentManager to guarantee the semaphore will  be released only once. And we have a new unit test for that.